File: Project Templates/Application Plug-in/Xcode Plugin.xctemplate/NSObject_Extension.m
//
//  NSObject_Extension.m
//  ___PACKAGENAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//


#import "NSObject_Extension.h"
#import "___PACKAGENAME___.h"

@implementation NSObject (Xcode_Plugin_Template_Extension)

+ (void)pluginDidLoad:(NSBundle *)plugin
{
    static dispatch_once_t onceToken;
    NSString *currentApplicationName = [[NSBundle mainBundle] infoDictionary][@"CFBundleName"];
    if ([currentApplicationName isEqual:@"Xcode"]) {
        dispatch_once(&onceToken, ^{
            sharedPlugin = [[___PACKAGENAME___ alloc] initWithBundle:plugin];
        });
    }
}
@end

File: Project Templates/Application Plug-in/Xcode Plugin.xctemplate/NSObject_Extension.swift
//
//  NSObject_Extension.swift
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import Foundation

extension NSObject {
    class func pluginDidLoad(bundle: NSBundle) {
        let appName = NSBundle.mainBundle().infoDictionary?["CFBundleName"] as? NSString
        if appName == "Xcode" {
        	if sharedPlugin == nil {
        		sharedPlugin = ___PACKAGENAME___(bundle: bundle)
        	}
        }
    }
}